_posts/2024-03-14-the-big-rename.adoc

```diff
@@ -0,0 +1,48 @@
+---
+layout: post
+title: 'Evolving Quarkus Extension Naming for Clarity'
+date: 2024-04-14
+tags: announcement
+synopsis: Recently, we've made a decision to rename several Quarkus extensions that previously included the word "reactive" in their names. This blog post aims to shed light on the reasoning behind this change and what it means for you as a user and/or extension writer.
+author: maxandersen
+---
+
+Quarkus is about providing a modern, efficient, and productive development experience for Java developers. We're committed to making it easier for you to build cloud-native applications, whether you're using traditional blocking, reactive, or virtual thread-based programming models.
+
+Historically Quarkus has used the term "reactive" in the names of several extensions to indicate their *additional* support for non-blocking, event-driven programming. 
+
+However, we've found that this terminology has led to confusion and misunderstanding among our users. Many have mistakenly believed that using these "reactive" extensions forces them into reactive programming, even though Quarkus always by default recommend traditional blocking programming model for your application and only optionally use reactive. 
+
+With the recent production release of Virtual Threads in Java 21, the distinction between traditional, reactive, and now, virtual thread-based programming has become even more critical. 
+
+== Our Solution: A New Naming Strategy
+
+To address this and guide our users more effectively, we've decided to revise the naming convention for certain extensions. This change aims to clarify the capabilities and flexibility of Quarkus extensions, making it easier for you to choose the right extension for your project.
+
+Here's a summary of the changes:
+
+- *Quarkus RESTEasy Reactive* becomes *Quarkus REST*, emphasizing its general-purpose applicability beyond just reactive programming.
+- *Quarkus SmallRye Reactive Messaging* is now *Quarkus Messaging*, indicating its broader use cases.
+- Extensions that inherently push for a reactive programming model, like *Quarkus Hibernate Reactive*, will retain the "reactive" terminology.
+- The *Quarkus REST Client Reactive* is simplified to *Quarkus REST Client*, unifying the naming under a more general "REST" umbrella.
+
```

Comment:
Quarkus REST, Quarkus Messaging and Quarkus REST Client are still leveraging the reactive engine underneath for efficiency and performances. At the user level, this is an implementation details.